wandb/sdk/wandb_settings.py

```diff
@@ -194,17 +178,65 @@ class Settings(six.with_metaclass(CantTouchThis, object)):
 
     """
 
-    # pylint: disable=no-member
-    entity: Optional[str]
+    mode: Optional[str] = None
+    console: Optional[str] = None
+    disabled: Optional[bool] = None
+    resume_fname_spec: Optional[str] = None
+    root_dir: Optional[str] = None
+    log_dir_spec: Optional[str] = None
+    log_user_spec: Optional[str] = None
+    log_internal_spec: Optional[str] = None
+    sync_file_spec: Optional[str] = None
+    sync_dir_spec: Optional[str] = None
+    files_dir_spec: Optional[str] = None
+    log_symlink_user_spec: Optional[str] = None
+    log_symlink_internal_spec: Optional[str] = None
+    sync_symlink_latest_spec: Optional[str] = None
+    settings_system_spec: Optional[str] = None
+    settings_workspace_spec: Optional[str] = None
+
+    __frozen: bool
+    __defaults_dict: Dict[str, int]
+    __override_dict: Dict[str, int]
+    __defaults_dict_set: Dict[str, Set[int]]
+    __override_dict_set: Dict[str, Set[int]]
+
+    class Source(object):
+        BASE: int = 0
+        ORG: int = 1
+        ENTITY: int = 2
+        PROJECT: int = 3
+        USER: int = 4
+        SYSTEM: int = 5
+        WORKSPACE: int = 6
+        ENV: int = 7
+        SETUP: int = 8
+        INIT: int = 9
+        SETTINGS: int = 10
+        ARGS: int = 11
```

Comment:
It would be nice to have a comment for each source describing where it comes from.

Reply:
fyi:
ORG = from the cloud on the org
PROJECT = from thee cloud on the project
...
SETUP = passed into wandb.step()
INIT = passed into wandb.init()
ARGS = passed as args to whatever method

I will write this all up.